src/main.rs

```diff
@@ -223,10 +237,56 @@ mod tests {
     }
 
     #[test]
-    fn test_hex(){
+    fn test_oct() {
+        assert_eq!(as_string_base(&4,   8).unwrap(), "4");
+        assert_eq!(as_string_base(&12,  8).unwrap(), "14");
+        assert_eq!(as_string_base(&187, 8).unwrap(), "273");
+        assert_eq!(as_string_base(&69,  8).unwrap(), "105");
+    }
+
+    #[test]
+    fn test_hex() {
         assert_eq!(as_string_base(&4,   16).unwrap(), "4");
         assert_eq!(as_string_base(&12,  16).unwrap(), "C");
         assert_eq!(as_string_base(&187, 16).unwrap(), "BB");
         assert_eq!(as_string_base(&69,  16).unwrap(), "45");
     }
+
+    #[test]
+    fn test_get_bases() {
+        let mut opt = Opt{
+            pad: 0,
+            sep_length: 4,
+            sep_char: '_',
+            no_sep: false,
+            from_base: 10,
+            silent: false,
+            bare: false,
+            verbosity: 0,
+            from_base_char: "b".to_owned(),
+            from_num: Some("187".to_owned()),
+            to_bases: Vec::new(),
+        };
+
+        let mut to_bases: Vec<String> = opt.to_bases.clone();
+        let res = get_bases(&opt, &mut to_bases);
+        assert_eq!(res.0, 2);
+        assert_eq!(res.1, Some("187".to_owned()));
+        assert!(to_bases.is_empty());
+
+        opt.from_base_char = "80".to_owned();
+        let res = get_bases(&opt, &mut to_bases);
+        assert_eq!(res.0, 10);
+        assert_eq!(res.1, Some("80".to_owned()));
+        assert!(!to_bases.is_empty());
+    }
+
+    #[test]
+    fn test_convert_to_base_10e() {
```

Comment:
Does the 'e' at the end of this function stand for something?
```suggestion
    fn test_convert_to_base_10() {
```

Reply:
oh sorry, yeah that's a typo, one second